app/src/main/kotlin/com/simplemobiletools/gallery/activities/ViewPagerActivity.kt

```diff
@@ -483,18 +496,45 @@ class ViewPagerActivity : SimpleActivity(), ViewPager.OnPageChangeListener, View
         }
     }
 
-    private fun saveFile(file: File, bitmap: Bitmap, out: OutputStream) {
+    private fun saveFile(file: File, bitmap: Bitmap, out: FileOutputStream) {
         val matrix = Matrix()
         matrix.postRotate(mRotationDegrees)
         val bmp = Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, true)
         bmp.compress(file.getCompressionFormat(), 90, out)
-        out.flush()
-        out.close()
-        toast(R.string.file_saved)
-        mRotationDegrees = 0f
-        invalidateOptionsMenu()
     }
 
+    private fun saveRotation(input: String, out: File) {
+        var inputStream: FileInputStream? = null
+        var outputStream: FileOutputStream? = null
+        try {
+            inputStream = FileInputStream(input)
+            outputStream = FileOutputStream(out)
+            inputStream.copyTo(outputStream)
+        } catch (ignored: FileNotFoundException) {
+        } finally {
+            inputStream?.close()
+            outputStream?.close()
+        }
+        if (mRotationDegrees != 0f) {
+            val exif = ExifInterface(out.absolutePath)
+            var orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
+            var orientationDegrees = when (orientation) {
+                8 -> 270f
+                3 -> 180f
+                6 -> 90f
+                else -> 0f
+            }
+            orientationDegrees = (orientationDegrees + mRotationDegrees) % 360
+            orientation = when (orientationDegrees) {
+                90f -> 6
+                180f -> 3
+                270f -> 8
+                else -> 1
+            }
```

Comment:
for these 2 "when" cases create some helper functions similar to https://github.com/SimpleMobileTools/Simple-Gallery/commit/3c3740731564b47b1be0e9c608b7852a2cc2eef2#diff-0c323285093b03e42d5770c0ff050227R260 , where instead of magic numbers 6,3,8,1 you use constants like ExifInterface.ORIENTATION_ROTATE_90 , degrees are ok. You can also cast orientation to string in the function, to avoid ugly code like: ""+ orientation

Reply:
- Haha I had that,but changed it to hhardcoded values for uniformity with https://github.com/SimpleMobileTools/Simple-Gallery/blob/3c3740731564b47b1be0e9c608b7852a2cc2eef2/app/src/main/kotlin/com/simplemobiletools/gallery/fragments/PhotoFragment.kt#L127

- I can make that function public to prevent redundancy or create a helper function, your call.
- Will also check if file is copied successfully beefore EXIF operation.
- Should I use an abstract copy function of yours maybe?

Reply:
just use android.media.ExifInterface for constants, they fit there.